journals/apps/search/backend.py

```diff
@@ -69,29 +76,13 @@ def get_mapping(self):
 
 class JournalsearchIndex(Elasticsearch5Index):
     '''Journal specific backend to Elasticsearch5'''
-    def put(self):
-        '''
-        Called during index creation
-        Override to setup the ingest_attachment plugin
-        '''
-        super(JournalsearchIndex, self).put()
-
-        # TODO is this the right place? Only do it once if it doesn't exist
-        if self.name == JOURNAL_DOCUMENT_INDEX_NAME:
-            body = {'description': 'Extract attachment information',
-                    'processors': [
-                        {INGEST_ATTACHMENT_ID: {'field': INGEST_ATTACHMENT_DATA_FIELD}}
-                    ]
-                    }
-
-            results = self.es.index(
-                index=INGEST_ATTACHMENT_INDEX,
-                doc_type=INGEST_ATTACHMENT_DOC_TYPE,
-                id=INGEST_ATTACHMENT_ID,
-                body=body
-            )
 
-            print('created index for _ingest attachment, results=', results)
+    def add_ingest_pipeline(self):
+        try:
+            self.es.ingest.get_pipeline(id=INGEST_ATTACHMENT_ID)
+        except NotFoundError:
+            results = self.es.ingest.put_pipeline(id=INGEST_ATTACHMENT_ID, body=INGEST_PIPELINE_BODY)
```

Comment:
also please wrap this with try/except and log the error using logger instead of print. also log the non-error result